tools/toolchains/arm.py

```diff
@@ -285,6 +285,10 @@ def correct_scatter_shebang(self, sc_fileref, cur_dir_name=None):
         """
         with open(sc_fileref.path, "r") as input:
             lines = input.readlines()
+            if getattr(self.target, "default_lib", "std") == "small":                                           
+                for index, line in enumerate(lines):                
+                    if "__MICROLIB" in line:                
+                        lines[index]=lines[index].replace(lines[index],"#define __MICROLIB"+"\n"+lines[index])                                        
```

Comment:
Can you explain why you need to do this?

Reply:
Removed this code and added the linker option via --predefine=D__MICROLIB